core/observers/STFLogger.cpp

```diff
@@ -8,8 +8,21 @@ namespace pegasus
                          PegasusState* state) :
         Observer((reg_width == 32) ? ObserverMode::RV32 : ObserverMode::RV64)
     {
-        // set up version and stf generation type
-        stf_writer_.open(filename);
+        if (filename.find(".stf", filename.length() - 5)
+            || filename.find(".zstf", filename.length() - 6)
+            || filename.find(".STF", filename.length() - 5)
+            || filename.find(".ZSTF", filename.length() - 6))
```

Comment:
The STF library is case-sensitive, so `.STF` and `.ZSTF` aren't valid extensions.